src/platforms/hosted/jlink_swd.c

```diff
@@ -101,8 +95,9 @@ bool jlink_swd_init(adiv5_debug_port_s *dp)
 
 	/* Set up the accelerated SWD functions for basic target operations */
 	dp->dp_low_write = jlink_adiv5_raw_write_no_check;
+	dp->dp_low_read = jlink_adiv5_raw_read_no_check;
 	dp->dp_read = firmware_swdp_read;
-	dp->error = jlink_adiv5_clear_error;
+	dp->error = firmware_swdp_error;
```

Comment:
This line can be dropped as this function is called via `bmda_swd_dp_init()` which runs just after the DP is already set up in the base ADIv5 implementation.